### ISSUE TITLE
NETSDK1206 and library projects

### ISSUE THREAD

**Issue by jaredpar** [dotnet/sdk#37826]
As part of upgrading to .NET 8 we're seeing NETSDK1206 warnings on our libraries and are unsure of how to proceed. These problems can be reproduced by doing the following: 

```cmd
> dotnet new classlib -n ExampleLib -o .
> dotnet add package SQLitePCLRaw.bundle_green -v 2.1.0
> dotnet build
```

This will produce the following warning: 

> C:\Program Files\dotnet\sdk\8.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(284,5): warning NETSDK1206: Found version-specific or di 
stribution-specific runtime identifier(s): alpine-arm, alpine-arm64, alpine-x64. Affected libraries: SQLitePCLRaw.lib.e_sqlite3. In .NET 8.0 and high 
er, assets for version-specific and distribution-specific runtime identifiers will not be found by default. See https://aka.ms/dotnet/rid-usage for d 
etails. [C:\Users\jaredpar\temp\classlib\ExampleLib.csproj]

This warning though is about native assets but this isn't a deployment or exe, it's just a library. If this were an exe I'd understand how this warning was relevant (even if I don't exactly know how to account for it). For libraries though I'm unsure why we get this warning or what actions we should be taking to resolve it. 

Can we get a bit of guidance about what library projects are supposed to do with this?

**Comment by baronfel** [comment]
For whoever triages, I've got a binlog available for this so ping me when you're doing this.

**Comment by jaredpar** [comment]
@baronfel no binlog needed, can be repro'd with the 3 easy steps above.

**Comment by baronfel** [comment]
Definite repro - I have a few questions for investigation:

* should the _build_ dependency file trigger this check, or just the `publish` dependency file?
* should `build` for libraries trigger this check at all?

Personal opinion: I think 'publish' workflows (so `publish`, but also `pack`) are a natural fit for this check, but I'm not sure that build is a good fit.

**Comment by jaredpar** [comment]
3 weeks since we filed this issue and no provided work arounds. We've had to just blanket suppress this warning in our repo.

**Comment by agocke** [comment]
Answer: the cause of this warning is because the SQLite NuGet package has authoring that will likely not work in .NET 8+. The important part from the link is:

> Starting with .NET 8, the default behavior of the .NET SDK and runtime is to only consider non-version-specific and non-distro-specific RIDs.

The listed RIDs there that mention a specific linux distribution (alpine) are non-portable. SQLite should update to not have alpine-specific RIDs.

Now, separately, to the question of what a *library* author should do: there's probably nothing you can do accept suppress the warning and bug the dependency for a new version. I think it would be reasonable for this warning to only show up for exes, not libraries. There is the problem that if you are build a Native AOT library, you are still effectively "the app". However, this feels like a corner case.

I think the questions about build vs publish are irrelevant. There's no meaningful difference between build and publish in this scenario.

**Comment by agocke** [comment]
Unfortunately I'm not familiar enough with MSBuild to answer that question. @dsplaisted?

**Comment by agocke** [comment]
FYI: SQLitePCLRaw has fixed this issue (and a bug on our side has been fixed) so this suppressing this warning may no longer be necessary after updating your references. See https://github.com/ericsink/SQLitePCL.raw/issues/543 for the fix.

**Comment by dsplaisted** [comment]
No, it doesn't look like we have a way to suppress this for a particular package.

**Comment by marcpopMSFT** [comment]
Old issue triage: sounds like the dependency updated for the new rid graph support and that was the only path forward. No way to suppress a warning for a single package. Closing out.
